test/sandbox/test/rendering_test.rb

```diff
@@ -418,10 +418,12 @@ def test_validations_component
   end
 
   def test_compiles_unrendered_component
+    skip "this might not still be compiled if it's been reloaded since boot"
     assert UnreferencedComponent.compiled?
   end
```

Comment:
Not sure if it's worth keeping this. UnreferencedComponent will get compiled on boot (because eager_load=true) but it might get reloaded before this test runs (because cache_classes=false).

I assume it's trying to test this line - https://github.com/ViewComponent/view_component/blob/964a2c15f74588819ea64ff69d12dfb0de80fe3b/lib/view_component/engine.rb#L68




If we reeaaally want to test that unreferenced components get compiled on boot, we could start a new rails process within the test:

```ruby
  def test_compiles_unrendered_component
    assert_equal "true\n", `ruby -r bundler/setup -r ./test/sandbox/config/environment.rb -e "puts UnreferencedComponent.compiled?"`
  end
```

Reply:
Since it's unreferenced, why would it get reloaded?

Reply:
It'll get eager-loaded during boot due to https://github.com/ViewComponent/view_component/blob/d6b385e0757e6fb1cebabc7f222fdf51a29be24c/test/sandbox/config/environments/test.rb#L45

and then reloaded whenever we call `app.reloader.reload!`